docs/source/configuration_file.md

```diff
@@ -0,0 +1,159 @@
+2. Other parameters that are needed for the scheduler of interest (e.g. `"base_lr": 1e-5, "max_lr": 1e-2` for `"CosineAnnealingLR"`).
+
+#### balance_samples
+Bool. Balance positive and negative labels in both the training and the validation datasets.
+
+#### mixup_alpha
+Float. Alpha parameter of the Beta distribution, see [original paper](https://arxiv.org/abs/1710.09412).
+
+#### transfer_learning
+- `retrain_model`: Filename of the pretrained model (`path/to/pretrained-model`). If `null`, no transfer learning is performed and the network is trained from scratch.
+- `retrain_fraction`: Float between 0. and 1. Controls the fraction of the pre-trained model that will be fine-tuned. For instance, if set to 0.5, then the second half of the model will be fine-tuned, while the first layers will be frozen.
+
+## Architecture
+Architectures for both segmentation and classification are available and described [here](models.rst).
+If the selected architecture is listed [here](../../ivadomed/loader/loader.py#L14), then a classification task is run, ie the ground-truth are labels extracted from `target`, instead of arrays for the segmentation task.
+
+### default_model (Mandatory)
+Define the default model (`Unet`) and mandatory parameters that are common to all available architectures (listed [here](models.rst)). If a tailored model is defined (see next section), then the default parameters are merged with the parameters that are specific to the tailored model.
+- `name`: `Unet` (default)
+- `dropout_rate`: float (e.g. 0.4).
+- `batch_norm_momentum`: float (e.g. 0.1).
+- `depth`: int, number of down-sampling operations.
+Note:
+- `in_channel` is automatically defined with `multichannel` and `contrast/training_validation`.
+- `out_channel` is automatically defined with `target_suffix`.
+
+### Tailored model (optional)
+Here are defined the tailored model and the parameters that are specific to it (ie not defined in the default model). See examples:
+- [FiLMedUnet](../../ivadomed/config/config.json#L64)
+    - `metadata`: choice between `"without"`, `"mri_params"`, and `"contrast"`. If `"mri_params"`, then vectors of [FlipAngle, EchoTime, RepetitionTime, Manufacturer] are input to the FiLM generator. If `"contrast"`, then image contrasts (according to `config/contrast_dct.json`) are input to the FiLM generator.
+- [HeMISUnet](../../ivadomed/config/config_spineGeHemis.json#L64)
+    - `missing_contrast`: Bool.
+- [UNet3D](../../ivadomed/config/config_tumorSeg.json#L65)
+    - `length_3D`: tuple indicating the size of the subvolumes or volume used for unet 3D model (depth, width, height).
+    - `padding_3D`: size of the overlapping per subvolume and dimensions (e.i `padding:0`). Note: In order to be used, each dimension of an input image needs to be a multiple of length plus 2 * padding and a multiple of 16. To change input image size use the following transformation `CenterCrop3D`. 
```

Comment:
i read it 3 times and i still don't understand what this parameter does. It is mentioned "size of the overlapping", but the parameter is called "padding", so is that an overlap (i.e. number of voxels, in each dimension, that overlap between two sliding windows), or a padding (i.e., number of voxels added in each dimension). 

Also, it is still unclear to me what a subvolume is.

"(e.i `padding:0`)" → did you mean "(e.g `padding:0`)"?

to clarify, i would maybe give a concrete example of a volume of size XxYxZ, and what would happen to the subvolume and to the padding applied to the subvolume

Reply:
Example provided [here](https://github.com/neuropoly/ivado-medical-imaging/blob/master/ivadomed/loader/loader.py#L423).
This parameter (and its behaviour) is likely to change is a very near future with PR #240 --> I am removing it for now and opening an issue to document it once refactored: issue #279.